drag_and_drop_v2/public/js/drag_and_drop_edit.js

```diff
@@ -183,12 +183,22 @@ function DragAndDropEditBlock(runtime, element) {
                         remove: function(e) {
                             var $el = $(e.currentTarget).closest('.zone-row'),
                                 classes = $el.attr('class'),
-                                id = classes.slice(classes.indexOf('zone-row') + 9);
+                                id = classes.slice(classes.indexOf('zone-row') + 9),
+                                // The id is of the form "zone-<index>".  We extract the index by
+                                // slicing off the first five letters and "casting" to integer.
+                                index = +id.slice(5),
+                                i;
```

Comment:
@bradenmacdonald @e-kolpakov  The code will be fragile anyway, and will break if you add a new class to the div.  But good point anyway, we can at least make the code we add less fragile, even though it doesn't help right now.
Regarding the loop variable, I intentionally declared it at the top since it is used after the loop as well.  I know it would be scoped to the function in any case, but coming from C++, I feel this way gets across the intention better -- it's not limited to the loop.  Also, https://github.com/edx/edx-platform/wiki/Javascript-standards-for-the-edx-platform#good-practices (though it doesn't mention loop variables explicitly).